src/components/StringFormatter/__tests__/StringFormatter.spec.js

```diff
@@ -3,34 +3,70 @@
  * @copyright IBM Security 2019
  */
 
-import { shallow } from 'enzyme';
+import { render } from '@testing-library/react';
 import React from 'react';
 
 import { StringFormatter } from '../../..';
-import value from '../_mocks_';
+import { namespace } from '../StringFormatter';
 
 describe('StringFormatter', () => {
-  let stringFormatter;
-
-  beforeEach(() => {
-    stringFormatter = shallow(<StringFormatter value={value} />);
+  test('should have no Axe or DAP violations', async () => {
+    const main = document.createElement('main');
+    render(
+      <StringFormatter
+        value="This is a long test string that would normally be truncated."
+        truncate
+        lines={2}
+        width="50px"
+      />,
+      {
+        // DAP requires a landmark '<main>' in the DOM:
+        container: document.body.appendChild(main),
+      }
+    );
+    await expect(document.body).toHaveNoAxeViolations();
+    await expect(document.body).toHaveNoDAPViolations('StringFormatter');
   });
 
-  it('renders', () => {
-    expect(stringFormatter).toMatchSnapshot();
+  test('should add a custom class', () => {
+    const { getByText } = render(
+      <StringFormatter value="test content" className="custom-class" />
+    );
+    expect(getByText(/test content/i)).toHaveClass('custom-class');
   });
 
-  it('renders the truncated variation', () => {
-    stringFormatter.setProps({ truncate: true });
+  test('should truncate text via the `truncate` prop', () => {
+    const { getAllByText } = render(
+      <StringFormatter value="test content" truncate />
+    );
+    // Expect two elements with the target text: the base component
+    // and the tooltip that is added when `truncate={true}`.
+    expect(getAllByText(/test content/i).length).toBe(2);
```

Comment:
I'm not sure if this will work, but another way to make the test to be more specific is to find the element by a selector and check that it has the provided text.

```js
expect(container.querySelector(`.${namespace}--truncate`).toHaveTextContent(/test content/i)
```

This way you can also confirm that where the text is located.

Reply:
Thanks! Just updated this test & removed some of the comments, since everything is a bit clearer in that block now.